### ISSUE TITLE
Autosave doesn't work, #{continuum_status} is empty

### ISSUE THREAD

**Issue by Teyras** [tmux-plugins/tmux-continuum#42]
I installed continuum by cloning it and then adding `run-shell ~/.tmux/plugins/tmux-continuum/continuum.tmux` to my config file. Resurrect works fine, but continuum doesn't seem to be doing anything. Also, the `#{continuum_status}` variable is empty. Is there any other thing I have to do?

**Comment by bruno-** [comment]
Did you try reloading tmux config file?

**Comment by Teyras** [comment]
Yes, in fact I waited for a couple of days and rebooted a handful of times.

**Comment by bruno-** [comment]
Can you please provide this info https://github.com/tmux-plugins/tmux-continuum/blob/master/CONTRIBUTING.md

**Comment by Teyras** [comment]
[.tmux.conf](https://gist.github.com/Teyras/c83d98f0b0227c3b3e9d1d6fa9405f37)
[last session file](https://gist.github.com/Teyras/4cf2dc0aa046293c9c6b397e1b8fb820)

**Comment by bruno-** [comment]
Can you try moving `run-shell` from [here](https://gist.github.com/Teyras/c83d98f0b0227c3b3e9d1d6fa9405f37#file-tmux-conf-L4) at the bottom of your `tmux.conf`.

In general you want to run your plugin after the options are set. In this case `@continuum-save-interval` is set **after** the plugin is run.

**Comment by Teyras** [comment]
Looks like autosave works, thanks! However, continuum_status is still empty.

**Comment by mdeguzis** [comment]
Variable is empty for me too

**Comment by fejiso** [comment]
Empty for me as well

**Comment by rrebollo** [comment]
Same problem for me here!!! Any ideas???

**Comment by clayreimann** [comment]
When I run the status command manually on my mac I see:
```shell
 ~ → ~/.tmux/plugins/tmux-continuum/scripts/continuum_status.sh
/Users/clayreimann/.tmux/plugins/tmux-continuum/scripts/continuum_status.sh: line 12: [: -gt: unary operator expected
off
```

**Comment by rpdelaney** [comment]
@clayreimann

Line 12 is: 
` if [ $save_int -gt 0 ]; then `

That error indicates that `$save_int` is unset when line 12 is reached. 

It's set here, on line 9: 
`local save_int="$(get_tmux_option "$auto_save_interval_option")"`

That explains why the error went away when you set `@continuum-save-interval`. It seems that `$auto_save_interval_option` needs to be populated with the default value in this script. 

I don't have this error and I'm not sure why not. But try changing line 9 to this: 
`local save_int="$(get_tmux_option "${auto_save_interval_option:-15}")"`

If that fixes it then a PR should be an easy fix.

**Comment by bestvibes** [comment]
Same issue for me as well! Has anyone found a solution yet? Setting `@continuum-save-interval` removes the error from the script but the status still shows up empty. I'm on Mac 10.14.5 and tmux 2.9a.

**Comment by rpdelaney** [comment]
@bestvibes Can you try the change I suggested in my comment above yours? I don't use this plugin anymore but if someone can confirm that it fixes the problem I will send a PR.

**Comment by bestvibes** [comment]
@rpdelaney I tried what you mentioned, and I had the same result as @clayreimann above your comment. The error went away when running the script and it just shows 15 now. However when I add the status to the status bar in tmux it still shows empty, even though the script returns 15. I hope that clarifies it! I can give more details tonight once I'm near my computer.

**Comment by slowkow** [comment]
My `#{continuum_status}` was `off` even though I had this line in my `.tmux.conf`:

```
set -g @continuum-restore 'on'
```

Then, I changed my `.tmux.conf` to include this line:

```
set -g @continuum-save-interval 15
```

Finally, I ran `tmux source ~/.tmux.conf` and the status changed to `15` instead of `off`.

Now it looks like it is working:

```
$ ls -T ~/.tmux/resurrect/
~/.tmux/resurrect
├── last -> tmux_resurrect_20191009T110430.txt
├── pane_contents.tar.gz
├── save
│  └── pane_contents
└── tmux_resurrect_20191009T110430.txt
```

I'm using tmux 2.9a on macOS.

**Comment by xave** [comment]
In addressing the empty status bar concern:

For manual installations, you need to make sure the `run-shell` command is after any `set -g` commands or the `Continuum-status:` will be blank, presumably because the script will have run before those optional settings have been provided to it otherwise. 

In essence:

    `set -g @foo`
    `set -g @bar`
    `run-shell /path/to/tmux/continuum/continuum.tmux` #must be the last line for continuum portion of tmux config

**Comment by alex-popov-tech** [comment]
Hi guys, same thing...
here is my tmux conf - https://github.com/alex-popov-tech/.dotfiles/blob/master/configs/tmux.conf
and i see empty continuum_status variable in status line, and automatic saving does NOT happens also :( 
but if i will run mayually `continuum_save.sh` or `continuum_status.sh` they both works as expected (save session or show current status text)

Any ideas on that?

cc @slowkow @rpdelaney

**Comment by alex-popov-tech** [comment]
@bruno- any updates on that topic?

**Comment by rpdelaney** [comment]
@alex-popov-tech The fix I described in my previous comment is still working for me. I'm not sure what else to suggest.

**Comment by illuz** [comment]
I put confg settings before `@plugin` and it works.
```
set -g @continuum-restore 'on'
set -g @resurrect-save-shell-history 'on'
set -g @continuum-save-interval '1'
```
Just try it. @slowkow @rpdelaney @alex-popov-tech

**Comment by rpdelaney** [comment]
> I don't have this error and I'm not sure why not. 

Please don't tag me... I can't help with this any more than I already have.

**Comment by mdeguzis** [comment]
Idk why it doesn't get sourced right, but this fixed it for me

```
# top of tmux.conf somewhere 
# For some reason, the var doesn't get set normally for #{continuum_status}
run-shell "tmux setenv -g continuum_status $(~/.tmux/plugins/tmux-continuum/scripts/continuum_status.sh)

if-shell '[ $(echo "$TMUX_VERSION >= 1.9" | bc -l) == 1 ]' \
    "\  
    set -g @plugin tmux-plugins/tpm; \
    set -g @plugin tmux-plugins/tmux-logging; \
    set -g @plugin tmux-plugins/tmux-resurrect; \
    set -g @plugin tmux-plugins/tmux-continuum; \
    set -g @continuum-restore 'on' \
    set -g @continuum-save-interval '15'; \
    set -g status-right 'Continuum status: #{continuum_status}'; \
    "
```

**Comment by danielkrajnik** [comment]
I think it was `set -g @continuum-save-interval '5';` that made it tick

**Comment by luchaoqi** [comment]
> I think it was `set -g @continuum-save-interval '5';` that made it tick

Same, need to specify the interval in the config file to `start` the plugin
